pkg/kernelmonitor/plugins/types/plugin.go

```diff
@@ -0,0 +1,45 @@
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package types
+
+import (
+	"io"
+
+	"k8s.io/node-problem-detector/pkg/kernelmonitor/types"
+)
+
+// Plugin is the interface of a log parsing plugin.
+type Plugin interface {
```

Comment:
The advantage of current interface is:
* Less code change, because current code is working like this.
* Plugin can share some code, such as the read loop.

However, delegating the whole logic to the plugin does seem to be more flexible to me:
* We can support kmsg, which has [different read logic](https://github.com/euank/node-problem-detector/blob/d834639cf9181850a408f6adde3c9154d4b9fb25/vendor/github.com/euank/go-kmsg-parser/kmsgparser/kmsgparser.go#L115-L130).
* We can use sdjournal api directly instead of the reader interface which has an extra buffer inside.

Comparing these 2 options, I prefer the later. :)

/cc @AdoHe WDYT?